src/System.Private.CoreLib/src/System/HighPerformanceCounter.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static class HighPerformanceCounter
+    {
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern ulong QueryPerformanceFrequency();
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern ulong QueryPerformanceCounter();
+
+        /// <summary>
+        /// Current tick count.
+        /// </summary>
+        /// <remarks>
+        /// Hardware used varies: (TSC, HPET, ACPI, proprietary timers)
+        /// On fairly new systems the API should be cheap, monotonic and synchronized.
+        /// In some fairly old cases, around Windows XP era, could be not synchronized between cores.
+        /// Could have relatively high latency on Windows 7 era environments.
+        /// </remarks>
+        public static ulong TickCount
+        {
+            get
+            {
+                return QueryPerformanceCounter();
+            }
+        }
```

Comment:
What is the benefit of this as compared to Stopwatch.GetTickCount(), just that it's not in corelib?

This represents the bulk of the Stopwatch implementation. If we really need it in corelib, then shouldn't we just move Stopwatch down rather than duplicating it?

And if we must duplicate it, can't we use the code (modified as necessary in the shared partition)?

I also don't like us adding code like this that's not actually used.

Reply:
Stopwatch is not in corlib, yes. It was recommended to look at CoreRt for how it deals with that. CoreRt has this. I am not sure why not Stopwatch. I assumed there are reasons for CoreRt having this API instead and also apply here. Like layering for additional features/diagnostics used there, or maybe just desire/rules to keep extension APIs out of corlib when possible. 
Surely the same questions came up for CoreRt and were resolved once already.

 I would be using this in an otherwise unrelated work. I did not want to bundle it all together. Also to manage risk of finding that the change is undesirable before investing more in this direction.

I kind of expected that this would be an uncontroversial simple change though, since I heard that not having HP timers in corlib was a problem in the past. And again - by looking at prior art in CoreRt.

Reply:
> was recommended to look at CoreRt for how it deals with that.

I have mentioned both CoreRT and Stopwatch as examples for how to deal with that. Both use regular PInvokes with signatures defined in shared partition. Neither of these two have duplicated FCalls.

> I assumed there are reasons for CoreRt having this API. 

We used to have a lot of duplication between CoreLib and CoreFX due to historical artificial layering. We do not have this artificial layering anymore and we prefer to avoid duplication.